From d4fa7f2ceee8786c3b0b482b8326f685027450ef Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Fri, 6 Jul 2012 22:11:35 +0800
Subject: ENGR00215968[MX6]Enable bus freq for i.MX6DL

Compared to i.MX6DQ, DL is not having high setpoint,
so just map the high setpoint to med setpoint for
DL.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-mx6/bus_freq.c |   16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/bus_freq.c b/arch/arm/mach-mx6/bus_freq.c
index cb4d6ea..2de1da2 100644
--- a/arch/arm/mach-mx6/bus_freq.c
+++ b/arch/arm/mach-mx6/bus_freq.c
@@ -255,6 +255,12 @@ int set_high_bus_freq(int high_bus_freq)
 	if (med_bus_freq_mode && !high_bus_freq)
 		return 0;
 
+	if (cpu_is_mx6dl() && high_bus_freq)
+		high_bus_freq = 0;
+
+	if (cpu_is_mx6dl() && med_bus_freq_mode)
+		return 0;
+
 	while (!mutex_trylock(&bus_freq_mutex))
 		msleep(1);
 
@@ -526,8 +532,13 @@ static int __devinit busfreq_probe(struct platform_device *pdev)
 
 	cpu_op_tbl = get_cpu_op(&cpu_op_nr);
 	low_bus_freq_mode = 0;
-	high_bus_freq_mode = 1;
-	med_bus_freq_mode = 0;
+	if (cpu_is_mx6dl()) {
+		high_bus_freq_mode = 0;
+		med_bus_freq_mode = 1;
+	} else {
+		high_bus_freq_mode = 1;
+		med_bus_freq_mode = 0;
+	}
 	bus_freq_scaling_is_active = 0;
 	bus_freq_scaling_initialized = 1;
 
@@ -586,7 +597,6 @@ static int __init busfreq_init(void)
 
 		printk(KERN_INFO "Bus freq driver Enabled\n");
 	}
-
 	return 0;
 }
 
-- 
1.7.9.5

